### DISCUSSION TITLE
POIs exported by poi-writer incomplete

### DISCUSSION THREAD

**Discussion by miki151** [mapsforge/mapsforge#1342] [General]
I've added a POI overlay to my map app with a search function, and I've noticed a lot of POIs in the database are missing. Below are a couple of screenshots where I searched for (1) the health care category and (2) all categories. You can clearly see that the map contains these items, but no POIs are present for them. I've tried with POI databases from the mapsforge download server as well as from openandromaps, and I've also tried exporting the POIs myself using osmosis and poi-writer:

bin/osmosis --rb file=~/Downloads/sweden-latest.osm.pbf --poi-writer file=sweden.poi ways=false names=false

Is there a way to retrieve a complete set of POIs from the original OSM data? Is there something I'm missing?

![signal-2022-07-22-154553_003](https://user-images.githubusercontent.com/5932991/180453482-721cbe91-b063-4a5a-bf99-dccfb4d5bfa3.png)
![signal-2022-07-22-154553_002](https://user-images.githubusercontent.com/5932991/180453488-f91e5a49-81eb-46d8-8204-c2b13cacee8a.png)

**Comment by devemux86** [comment]
POI creation is described in the [documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/POI.md) using a poi-mapping xml.
(OpenAndroMaps uses a similar configuration file)

However some POI may be on (closed) ways.

Map files are not related with POI files, they use separate xml configurations.

Better compare the OSM tags for nodes + ways for maps and pois.

**Reply by miki151** [reply]
I think that a hospital POI should be present in the database though. I assumed that all the little icons in OSM would be extracted as POIs, I guess I'll have to look for another source then.

**Comment by devemux86** [comment]
> I think that a hospital POI should be present in the database though.

They are [included](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml#L121).
However Mapsforge include the [named POI](https://github.com/mapsforge/mapsforge/blob/master/docs/POI.md#advanced-options-only-use-when-you-know-what-you-are-doing).

You can create you own poi files, but use the default [ways=true](https://github.com/mapsforge/mapsforge/blob/master/docs/POI.md#advanced-options-only-use-when-you-know-what-you-are-doing).

**Reply by miki151** [reply]
I generated my own database with unnamed POIs as well and the hospital wasn't there. With ways=true, I get a sqlite crash due to too big DB, but I'll try to work around that. Is it possible that hospitals are part of ways, anyway?

**Comment by devemux86** [comment]
> Is it possible that hospitals are part of ways, anyway?

You can check the OpenStreetMap to see how they are declared.

In maps many poi symbols / labels are set on the center of (closed) ways, i.e. buildings.

**Reply by miki151** [reply]
Alright, I generated a database with ways=true and names=false, and the playgrounds have appeared, but still no hospital. I can see all the building numbers, parking lots, etc, but no actual point with the hospital name, neither in Healthcare category nor any other.

**Comment by devemux86** [comment]
Post a specific example for a small map / poi file at **Andorra** that anyone can easily check.

Note that the map may render **multiple same** symbols / labels on building multi-polygons.
So the map _may appear_ to contain more elements (it does not).

If map-writer / poi-writer options are correct, comparison must be done on **unique** elements.
(not compared to map rendering)
